### ISSUE TITLE
IAN模型的实现

### ISSUE THREAD

**Issue by chosenone75** [songyouwei/ABSA-PyTorch#11]
关于IAN模型的实现，

        

> aspect_len = torch.tensor(aspect_len, dtype=torch.float).to(self.opt.device)
>         aspect = torch.sum(aspect, dim=1)
>         aspect = torch.div(aspect, aspect_len.view(aspect_len.size(0), 1))
>         text_raw_len = torch.tensor(text_raw_len, dtype=torch.float).to(self.opt.device)
>         context = torch.sum(context, dim=1)
>         context = torch.div(context, text_raw_len.view(text_raw_len.size(0), 1))
>         aspect_final = self.attention_aspect(aspect, context).squeeze(dim=1)
>         context_final = self.attention_context(context, aspect).squeeze(dim=1)

在上面的实现中，`context `和`aspect `最后均为平均值，是否与原文模型不符。

**Comment by songyouwei** [comment]
@chosenone75 IAN的实现应该是正确的。计算hidden state，平均，计算interactive attention，拼接，全连接。
paper第二页：
> Then, we can get the initial representations of context and target (i.e., cavg and tavg ) by averaging the hidden states.

**Comment by chosenone75** [comment]
这里最后的 context 和 aspect 的值都是平均后的（被覆盖了），作为 attention layer的输入是不是不太对？还是我理解有问题？

**Comment by songyouwei** [comment]
@chosenone75 确实覆盖了。。确实写错了😂
